File: app/src/main/java/com/recursia/popularmovies/domain/SearchScreenInteractorImpl.java
package com.recursia.popularmovies.domain;

import com.recursia.popularmovies.domain.models.Movie;

import java.util.List;

import io.reactivex.Single;

public class SearchScreenInteractorImpl implements SearchScreenInteractor {

    private final MoviesRepository moviesRepository;

    public SearchScreenInteractorImpl(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    @Override
    public Single<List<Movie>> getMoviesByQuery(String query, int page, String language) {
        return moviesRepository.getMoviesByQuery(query, page, language);
    }

}

File: app/src/main/java/com/recursia/popularmovies/domain/SearchScreenInteractorImpl.kt
package com.recursia.popularmovies.domain

import com.recursia.popularmovies.domain.models.Movie

import io.reactivex.Single

class SearchScreenInteractorImpl(private val moviesRepository: MoviesRepository) : SearchScreenInteractor {

    override fun getMoviesByQuery(query: String, page: Int, language: String): Single<List<Movie>> {
        return moviesRepository.getMoviesByQuery(query, page, language)
    }
}
